game_logic.cpp

```diff
@@ -0,0 +1,118 @@
+#include "game_logic.h"
+
+//gets input from console if player is human
+//generates random input if player is bot
+std::string getPlayerInput(Board &board, Player player, int askCoordinate = 0){
+    std::string ret = "";
+    if(player.isAI == false){
+        std::cin >> ret;
+        return ret;
+    }
+    else{
+        std::vector<Position> validTiles = getValidTiles(board); //TODO: make bot not be able to choose the positions where their own mines are placed
+        int randomPos = rand() % validTiles.size();

```

Comment:
#should pass random generation function as a parameter instead of directly calling `rand()` here